internal/services/network/subnet_resource.go

```diff
@@ -25,39 +25,64 @@ import (
 var SubnetResourceName = "azurerm_subnet"
 
 var subnetDelegationServiceNames = []string{
+	"GitHub.Network/networkSettings",
 	"Microsoft.ApiManagement/service",
+	"Microsoft.Apollo/npu",
 	"Microsoft.App/environments",
+	"Microsoft.App/testClients",
+	"Microsoft.AVS/PrivateClouds",
 	"Microsoft.AzureCosmosDB/clusters",
 	"Microsoft.BareMetal/AzureHostedService",
+	"Microsoft.BareMetal/AzureHPC",
+	"Microsoft.BareMetal/AzurePaymentHSM",
 	"Microsoft.BareMetal/AzureVMware",
 	"Microsoft.BareMetal/CrayServers",
+	"Microsoft.BareMetal/MonitoringServers",
 	"Microsoft.Batch/batchAccounts",
+	"Microsoft.CloudTest/hostedpools",
+	"Microsoft.CloudTest/images",
+	"Microsoft.CloudTest/pools",
+	"Microsoft.Codespaces/plans",
 	"Microsoft.ContainerInstance/containerGroups",
 	"Microsoft.ContainerService/managedClusters",
+	"Microsoft.ContainerService/TestClients",
 	"Microsoft.Databricks/workspaces",
 	"Microsoft.DBforMySQL/flexibleServers",
+	"Microsoft.DBforMySQL/servers",
 	"Microsoft.DBforMySQL/serversv2",
 	"Microsoft.DBforPostgreSQL/flexibleServers",
 	"Microsoft.DBforPostgreSQL/serversv2",
 	"Microsoft.DBforPostgreSQL/singleServers",
+	"Microsoft.DelegatedNetwork/controller",
+	"Microsoft.DevCenter/networkConnection",
+	"Microsoft.DocumentDB/cassandraClusters",
+	"Microsoft.Fidalgo/networkSettings",
 	"Microsoft.HardwareSecurityModules/dedicatedHSMs",
 	"Microsoft.Kusto/clusters",
-	"Microsoft.Logic/integrationServiceEnvironments",
 	"Microsoft.LabServices/labplans",
+	"Microsoft.Logic/integrationServiceEnvironments",
 	"Microsoft.MachineLearningServices/workspaces",
 	"Microsoft.Netapp/volumes",
 	"Microsoft.Network/dnsResolvers",
-	"Microsoft.Network/managedResolvers",
+	"Microsoft.Network/fpgaNetworkInterfaces",
+	"Microsoft.Network/networkWatchers.",
+	"Microsoft.Network/virtualNetworkGateways",
+	"Microsoft.Orbital/orbitalGateways",
+	"Microsoft.PowerPlatform/enterprisePolicies",
 	"Microsoft.PowerPlatform/vnetaccesslinks",
 	"Microsoft.ServiceFabricMesh/networks",
+	"Microsoft.ServiceNetworking/trafficControllers",
+	"Microsoft.Singularity/accounts/networks",
+	"Microsoft.Singularity/accounts/npu",
 	"Microsoft.Sql/managedInstances",
-	"Microsoft.Sql/servers",
```

Comment:
same here

Reply:
re-added

Reply:
all though, my original error was `Message="The service name Microsoft.Sql/servers specified on delegation /subscriptions/xxxxx-xxxx-xxxx-xxxx-xxxxxxx/resourceGroups/myorg-connectivity-northeurope/providers/Microsoft.Network/virtualNetworks/myorg-hub-northeurope/subnets/mysubnet/delegations/Microsoft.Sql/servers is not valid.`

Reply:
changing the validation may break users configs that would otherwise be fine as terraform may not try to apply them (or its still valid for their instance somehow), therefore we always try to minimize the chance of this and do such changes in major versions